intg-appletv/tv.py

```diff
@@ -514,6 +533,59 @@ async def _update_app_list(self) -> None:
 
         self.events.emit(EVENTS.UPDATE, self._device.identifier, update)
 
+    async def _update_output_devices(self) -> None:
+        _LOG.debug("[%s] Updating available output devices list", self.log_id)
+        current_output_devices = self._available_output_devices
+        current_output_device = self.output_device
+        self._available_output_devices = {}
+        device_ids = []
+        try:
+            atvs = await pyatv.scan(self._loop)
+            for atv in atvs:
+                if atv.device_info.output_device_id == self._atv.device_info.output_device_id:
```

Comment:
While testing I got the following runtime exception:
```
  File ".../intg-appletv/tv.py", line 545, in _update_output_devices
    if atv.device_info.output_device_id == self._atv.device_info.output_device_id:
                                           ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'device_info'
```

`self._atv` is None if not connected!

Reply:
This is strange because the caller (connect method) also uses self._atv earlier. Anyway I added a check